src/snapred/backend/data/GroceryService.py

```diff
@@ -768,100 +772,264 @@ def fetchNeutronDataSingleUse(self, runNumber: str, useLiteMode: bool, loader: s
         :rtype: Dict[str, Any]
         """
 
-        filename: str = self._createNeutronFilename(runNumber, useLiteMode)
+        runNumber, useLiteMode, loader, liveDataArgs = item.runNumber, item.useLiteMode, item.loader, item.liveDataArgs
+
+        # Live-data mode can be requested explicitly,
+        #   but it also serves as the fallback mode.
+        liveDataMode = loader == "LoadLiveData" or liveDataArgs is not None
+
         workspaceName: str = self._createNeutronWorkspaceName(runNumber, useLiteMode)
 
+        success = False
+        convertToLiteMode = False
+
         self._updateNeutronCacheFromADS(runNumber, useLiteMode)
 
-        # check if a raw workspace exists, and clone it if so
         if self._loadedRuns.get(self._key(runNumber, useLiteMode)) is not None:
+            # a cached copy exists: clone it
             self.getCloneOfWorkspace(self._createRawNeutronWorkspaceName(runNumber, useLiteMode), workspaceName)
             data = {
                 "result": True,
                 "loader": "cached",
                 "workspace": workspaceName,
             }
-        # otherwise fetch the data
-        else:
-            data = self.grocer.executeRecipe(filename, workspaceName, loader)
+            success = True
+
+        if not success and not liveDataMode:
+            liteModeFilePath: Path = self._createNeutronFilePath(runNumber, True)
+            nativeModeFilePath: Path = self._createNeutronFilePath(runNumber, False)
+            cachedNativeMode = self._loadedRuns.get(self._key(runNumber, False)) is not None
+
+            match (useLiteMode, cachedNativeMode, liteModeFilePath.exists(), nativeModeFilePath.exists()):
+                # THERE ARE MANY SPECIAL CASES HERE, but basically a single-use fetch will use
+                #   cached (or on-disk lite versions) if available, but will not create any new cache entries if not.
```

Comment:
I had actually taken a look at this as part of a noncritical issue that had gotten shelved and did some work to break it up into its component parts in [this branch here](https://github.com/neutrons/SNAPRed/blob/ewm7732_normalize_by_direct_beam_monitor/src/snapred/backend/data/GroceryService.py#L750-L791), I think it simplifies the logic significantly.